packages/web-components/src/menu/menu.ts

```diff
@@ -154,7 +156,17 @@ export class Menu extends FASTElement {
    * Closes the menu.
    * @public
    */
-  public closeMenu = () => {
+  public closeMenu = (event?: Event) => {
+    // Keep menu open if the event target is a menu item checkbox or radio
+    if (
+      event &&
+      event.target instanceof MenuItem &&
+      (event.target.getAttribute('role') === MenuItemRole.menuitemcheckbox ||
+        event.target.getAttribute('role') === MenuItemRole.menuitemradio)
```

Comment:
👍  Right on. This is my preferred way to avoid `||` statements like this, especially when there's an `&&` in there. Just didn't know how everyone else felt. Will update.

Reply:
We could go even deeper:

```ts
const role = (event?.target as HTMLElement)?.getAttribute('role') ?? '';
if (role in CheckableMenuItemRole) {
  return;
}
```

and in `menu-item.options.ts`, add:

```ts
export const CheckableMenuItemRole = {
  [MenuItemRole.menuitemcheckbox]: 'menuitemcheckbox',
  [MenuItemRole.menuitemradio]: 'menuitemradio',
} as const;

export type CheckableMenuItemRole = ValuesOf<typeof CheckableMenuItemRole>;
```

Reply:
Update: TypeScript hated my attempt to use `Array.includes`. I think it sees two `MenuItemRole` items in the array and infers the type and `event?.target.getAttribute('role')` returns `string | null` so it panics with the following even though there's no assignment:

```
Argument of type 'string | null' is not assignable to parameter of type '"menuitemcheckbox" | "menuitemradio"'.
  Type 'null' is not assignable to type '"menuitemcheckbox" | "menuitemradio"'
```

In an effort to avoid using more TypeScript to fix TypeScript, I left it as the `||` conditional but did clean it up some:

```js
if (
  event?.target instanceof MenuItem &&
  (event.target.getAttribute('role') === MenuItemRole.menuitemcheckbox ||
    event.target.getAttribute('role') === MenuItemRole.menuitemradio)
) {
  return;
}
```